### PR TITLE
ROX-10079: oss helm charts publishing

### PR DESCRIPTION
**Pull request by jschnath**
## Description

So far we publish helm charts for `rhacs`and `stackrox.io` flavors, this adds helm charts for `opensource` flavor. Unlike the other two we want to package these charts in .tgz archives with helm publish in order to be able to point `helm repo add` at them.

The charts should be published at "https://github.com/stackrox/helm-charts/opensource/" in the correct way so that `helm repo add`works.

## Checklist
- [x] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

## Testing Performed

I'll have to do some manual testing to ensure that helm repo add works here I think. A working example of helm chart publishing to compare to can be found at https://github.com/stackrox/release-artifacts/blob/main/scripts/publish-helm-charts.sh

## TODO
ROX-10079 further asks for creating [artifacthub.io](https://artifacthub.io/ ) entries for these charts, I will look at this together with Matthias on Monday.

### CODE REVIEW

**Review by msugakov — COMMENTED** [review]
Code-wise the change looks good and it's great that you seem figured out all needed commands.
However, the problem is: how do you know it actually works? Remember that the goal is to allow user do `helm repo add` to add our GitHub opensource charts.
I suggest that you try execute all necessary `roxctl helm output`, `git` and `helm package`, `helm repo index` command manually to create an opensource chart tarball (it's ok to do for non-release version at this point), commit and push it to helm-charts repository and then try `helm repo add` and even install StackRox from it.

Once this works, you will be able to:
1. Tell confidently that this PR works.
2. Perhaps will refine commands in this PR in case something wasn't correct on the first attempt.
3. You'll have installation instructions that you'll be able to put in the README.md file for community users to follow (there's no JIRA task for that yet, but it's needed anyway).

Later on, you can push git revert to remove that testing chart from `helm-charts` repo.

File: .circleci/publish-helm-charts [hunk 1]
```diff
@@ -46,6 +46,13 @@ mkdir "${tmp_remote_repository}/${remote_subdirectory}/rhacs/${version}"
 cp -a "${central_services_chart}/rhacs" "${tmp_remote_repository}/${remote_subdirectory}/rhacs/${version}/central-services"
 cp -a "${secured_cluster_services_chart}/rhacs" "${tmp_remote_repository}/${remote_subdirectory}/rhacs/${version}/secured-cluster-services"
 
+mkdir "${tmp_remote_repository}/${remote_subdirectory}/opensource"
```
[the remark this answers is not in the document]
Comment (jschnath) [reply .circleci/publish-helm-charts]: done

File: .circleci/publish-helm-charts [hunk 2]
```diff
@@ -46,6 +46,13 @@ mkdir "${tmp_remote_repository}/${remote_subdirectory}/rhacs/${version}"
...
+
+helm package -d "${tmp_remote_repository}/${remote_subdirectory}/opensource" "${central_services_chart}/opensource/Chart.yaml" || die "Failed to package Helm chart for file ${central_services_chart}/opensource/Chart.yaml"
+helm package -d "${tmp_remote_repository}/${remote_subdirectory}/opensource" "${secured_cluster_services_chart}/opensource/Chart.yaml" || die "Failed to package Helm chart for file ${secured_cluster_services_chart}/opensource/Chart.yaml"
+
+helm repo index "${tmp_remote_repository}/${remote_subdirectory}/opensource" || die "Failed to build helm index"
```
[the remark this answers is not in the document]
Comment (jschnath) [reply .circleci/publish-helm-charts]: done

**Comment by jschnath** [comment]
Testing the commands manually right now

**Comment by jschnath** [comment]
Successfully installed the helm chart using `helm install stackrox-central-services https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource//stackrox-central-services-70.0.121-g80ac8b48db.tgz --debug -n stackrox` and deployed it locally to docker, looks good.

**Comment by msugakov** [comment]
```sh
helm repo update
```
is redundant, it seems.

Instead of providing URL (`https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource//stackrox-central-services-70.0.121-g80ac8b48db.tgz`) to the tarball, I used the following command to deploy Central
```sh
helm install -n stackrox --create-namespace stackrox-central-services stackroxhelm/stackrox-central-services --devel
```

I suppose you'd use some similar command to document installation approach for OSS/community users.

File: pkg/images/defaults/flavor.go [hunk 3]
```diff
@@ -218,8 +218,7 @@ func OpenSourceImageFlavor() ImageFlavor {
 		ScannerDBSlimImageName: "scanner-db-slim",
 
 		ChartRepo: ChartRepo{
-			// TODO(ROX-10079): establish a place where open source users will be able to download Helm charts.
-			URL: "https://charts.stackrox.io",
+			URL: "https://github.com/stackrox/helm-charts/opensource/",
```
Comment (msugakov) [inline pkg/images/defaults/flavor.go:221]: This needs to change now as we're using ...raw.githubusercontent...
To see why it matters, how and where it is used, you can do the following:
1. Run `make cli`, unless you've already done so.
2. Run `roxctl helm output central-services --image-defaults opensource --output-dir blah`.
3. Examine `blah/README.md` file produced by the last command. You'll find mention of `helm repo add` and this URL there.
 
See commands you used in testing to figure out the proper URL to put.

**Comment by jschnath** [comment]
Documentation is a good point. Where should I add documentation for open source users? I'm thinking it should go somewhere here https://docs.openshift.com/acs/3.69/installing/installing_helm/install-helm-quick.html since this is linked to on [stackrox.io](https://www.stackrox.io/docs/), but I don't think I have write access there.

File: .circleci/publish-helm-charts [hunk 4]
```diff
@@ -46,6 +46,16 @@ mkdir "${tmp_remote_repository}/${remote_subdirectory}/rhacs/${version}"
...
 
+mkdir -p "${tmp_remote_repository}/${remote_subdirectory}/opensource"
+
+echo "Packaging Helm chart for file ${central_services_chart}/opensource/Chart.yaml"
+helm package -d "${tmp_remote_repository}/${remote_subdirectory}/opensource" "${central_services_chart}/opensource"
+echo "Packaging Helm chart for file ${secured_cluster_services_chart}/opensource/Chart.yaml"
+helm package -d "${tmp_remote_repository}/${remote_subdirectory}/opensource" "${secured_cluster_services_chart}/opensource"
+
+echo "Building OSS helm repo index"
+helm repo index "${tmp_remote_repository}/${remote_subdirectory}/opensource"
```
Comment (msugakov) [inline .circleci/publish-helm-charts:57]: As for testing code in this file, have you tried running it? Does it successfully execute these new commands?
You can comment-out `git push` and `curl` commands below and run this script to see if produced changes in the repo are what you expect.

**Comment by msugakov** [comment]
Please remember that stackrox.io is for **commercial** users, not **community** users. Yes, there's confusion.

I don't think that openshift docs is a good location because it is again for commercial users.

I think this documentation should go somewhere to README.md file in the root of this repo. Remember, our goal is to make entry barrier as low as possible for community users. This README can follow a typical pattern of open source projects such as first introduce the product (you don't have to write that), next explain how to install it (your part), next explain how to develop for it (currently existing content).
I think, it is best to create a separate JIRA for writing this documentation (unless there's one already).

File: pkg/images/defaults/flavor.go (hunk 3 above)
Comment (jschnath) [reply pkg/images/defaults/flavor.go:221]: Makes sense, I'll test it but I expect the correct repo link to be `https://raw.githubusercontent.com/stackrox/helm-charts/main/opensource/` then

File: .circleci/publish-helm-charts (hunk 4 above)
Comment (jschnath) [reply .circleci/publish-helm-charts:57]: The commands here are the same ones I ran manually to create the helm charts I pushed to helm-charts/opensource, but I'll run the test you suggested as well to double check.

File: pkg/images/defaults/flavor.go (hunk 3 above)
Comment (jschnath) [reply pkg/images/defaults/flavor.go:221]: Yeah looks good.

**Comment by jschnath** [comment]
After manual testing this all seems good to me. I've also opened a separate issue for documentation at https://issues.redhat.com/browse/ROX-11172

**Review by msugakov — APPROVED** [review]

**Pull request merged**